master/buildbot/status/web/base.py

```diff
@@ -403,6 +403,68 @@ def path_to_json_past_builds(request, builderName, number, filter_data=False):
     return url
 
 
+def get_query_branches_for_codebases(tags, codebases):
+    regex_branches = [
+        r'^(trunk)/',                 # Trunk
+        r'^(20[0-9][0-9].[0-9])\/',   # 2017.1/
+        r'^([0-9].[0-9])\/',          # 5.0/
+        r'^release\/([0-9].[0-9])/'   # release/4.6
+    ]
```

Comment:
These regexes exist somewhere else in the code right?

Can we reuse it instead of duplicating?

Reply:
Yes, they are in rtBuilders.js.

But I don't use rtBuilders.js on builder-page, so I could do filtering by 2 ways
1. Run rtBuilders.js logic on builder-page and adjust a JS stack for these showing
2. Copy logic from JS to Python, and print builder-tags by simple Jinja

I used the second approach.

Reply:
Copying is bad, is there a way to have the Python code obtain the regexes from JS or vice-versa?

At the very least we should have a test somewhere to ensure that they are the same.

Reply:
Ok. I am working on using regexp declared in python file in JS.